From 8a21b58b45edbf11d9041f884bb9c42e6440104c Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Tue, 26 Dec 2023 14:38:58 +0100
Subject: [PATCH] libsoup/{soup-xmlrpc,soup-xmlrpc-old}: add missing
 <libxml/parser.h> include

Since libxml2 2.12.1, including <libxml/parser.h> is needed to get the
prototype of xmlParseMemory(), causing a build breakage of libsoup
with a recent version of libxml2.

Upstream: Not Applicable
[libsoup2 is not maintained upstream, and libsoup3 no longer contains
the XMLRPC code. We are keepin libsoup2 in Buildroot for compatibility
with packages that haven't moved to libsoup3 yet.]
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 libsoup/soup-xmlrpc-old.c | 1 +
 libsoup/soup-xmlrpc.c     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/libsoup/soup-xmlrpc-old.c b/libsoup/soup-xmlrpc-old.c
index c57086b6..dd51de53 100644
--- a/libsoup/soup-xmlrpc-old.c
+++ b/libsoup/soup-xmlrpc-old.c
@@ -12,6 +12,7 @@
 #include <string.h>
 
 #include <libxml/tree.h>
+#include <libxml/parser.h>
 
 #include "soup-xmlrpc-old.h"
 #include "soup.h"
diff --git a/libsoup/soup-xmlrpc.c b/libsoup/soup-xmlrpc.c
index 42dcda9c..b0132d26 100644
--- a/libsoup/soup-xmlrpc.c
+++ b/libsoup/soup-xmlrpc.c
@@ -18,6 +18,7 @@
 #include <string.h>
 #include <errno.h>
 #include <libxml/tree.h>
+#include <libxml/parser.h>
 #include "soup-xmlrpc.h"
 #include "soup.h"
 
-- 
2.43.0

